rllib/utils/tf_utils.py

```diff
@@ -202,18 +205,20 @@ def get_placeholder(
     if space is not None:
         if isinstance(space, (gym.spaces.Dict, gym.spaces.Tuple)):
             if flatten:
-                return ModelCatalog.get_action_placeholder(space, None)
+                return ModelCatalog.get_action_placeholder(space, None, one_hot=one_hot)
             else:
                 return tree.map_structure_with_path(
                     lambda path, component: get_placeholder(
                         space=component,
                         name=name + "." + ".".join([str(p) for p in path]),
+                        one_hot=one_hot,
                     ),
                     get_base_struct_from_space(space),
                 )
+        _, shape = ModelCatalog.get_action_shape(space, framework="tf", one_hot=one_hot)
         return tf1.placeholder(
-            shape=(None,) + ((None,) if time_axis else ()) + space.shape,
-            dtype=tf.float32 if space.dtype == np.float64 else space.dtype,
+            shape=(None,) + ((None,) if time_axis else ()) + shape[1:],
```

Comment:
why are we skipping that first index of that shape returned from the action?

Reply:
Great question. We are now getting the shape from the call to `_, shape = ModelCatalog.get_action_shape(space, framework="tf", one_hot=one_hot)` instead of from the space directly. That's why we have to cut the batch dim (which was not part of the space shape before, but it is part of the shape returned by `get_action_shape`).